### ISSUE TITLE
Add transformer reconstruction loss

### ISSUE THREAD

**Issue by jbloomAus** [ai-safety-foundation/sparse_autoencoder#89]
See [here](https://transformer-circuits.pub/2023/monosemantic-features#:~:text=We%20would%20like,difference%20in%20loss.)

To do:
- [ ] Consider the parameters needed to configure this (frequency of measurement, how much data to test on)
- [ ] Write the code run the full model with / without the SAE. 
- [ ] Log the loss recovery. 

Extension:
- [ ] Log any data points that the SAE causes especially poor performance on and the general distribution of per token loss.
- [ ] Measure slow down in training  for having calculated this

**Comment by alan-cooney** [comment]
To note this should probably be added to the pipeline validation stage as a metric, not the train stage. As such we don't have to be that worried about performance, and it can be used in hyperparam sweeps still but not as a direct loss value.
